### PR TITLE
Add lookup endpoint for community videos

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: app/commands/community_video/retrieve.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        CommunityVideo::RetrieveFromVimeo.(url)
+      else
+        raise InvalidCommunityVideoUrl
+      end
+    end
+
+    def youtube?
+      return true if url.starts_with?('https://youtu.be/')
+      return true if url.starts_with?('https://youtube.com')
+      return true if url.starts_with?(%r{https://[a-z]*\.?youtube.com/})
```
Comment (ErikSchierboom) [inline app/commands/community_video/retrieve.rb]: Should this be:

```suggestion
      return true if url.starts_with?(%r{https://[a-z]+\.youtube.com/})
```

The case without a subdomain is already caught above.

File: app/commands/community_video/retrieve_from_youtube.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        title: snippet["title"],
+        watch_id: youtube_id,
+        embed_id: youtube_id,
+
+        channel_name: snippet["channelTitle"],
+        thumbnail_url: snippet["thumbnails"]["standard"]["url"]
+      )
+      # rescue
+      #   raise InvalidCommunityVideoUrl
```
Comment (ErikSchierboom) [inline app/commands/community_video/retrieve_from_youtube.rb]: There is no TODO here. Did you want to re-enable this?

File: app/controllers/tracks/approaches_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Tracks::ApproachesController < ApplicationController
+  include UseTrackExerciseSolutionConcern
+  before_action :use_solution
+
+  skip_before_action :authenticate_user!
+
+  def index
+    #  TODO: Erik - only approved
```
Comment (ErikSchierboom) [inline app/controllers/tracks/approaches_controller.rb]: Can this be removed? It looks like only approved videos are returned.

```suggestion
```

**Review by ErikSchierboom — APPROVED** [review]

File: app/controllers/api/community_videos_controller.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+module API
+  class CommunityVideosController < BaseController
+    def lookup
+      video = CommunityVideo::Retrieve.(params[:video_url])
+      serialized = video.attributes.slice(*%w[title platform channel_name thumbnail_url])
```
Comment (ErikSchierboom) [inline app/controllers/api/community_videos_controller.rb]: I know Aron need to use the URL in the UI (it was fixed in the other branch)

```suggestion
      serialized = video.attributes.slice(*%w[title platform channel_name thumbnail_url url])
```

**Review by dem4ron — APPROVED** [review]

File: app/javascript/components/modals/upload-video/UploadVideoModal.tsx [hunk 5]
```diff
@@ -6,51 +6,76 @@ import {
...
     setVideoRetrievalFailure(false)
     setVideoData(null)
   }, [])
 
+  function renderUploadSteps() {
+    switch (videoUploadStep) {
+      case 'RETRIEVE':
+        return (
+          <>
+            <UploadVideoModalHeader videoSubmitted={false} />
```
Comment (dem4ron) [inline app/javascript/components/modals/upload-video/UploadVideoModal.tsx]: this should be `videoRetrieved={false}`

Comment (ErikSchierboom) [reply app/javascript/components/modals/upload-video/UploadVideoModal.tsx]:
```suggestion
            <UploadVideoModalHeader videoRetrieved={false} />
```

File: app/javascript/components/modals/upload-video/elements/UploadVideoForm.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      <UploadVideoTextInput
+        name="title"
+        label="Video title"
+        className="mb-24"
+        defaultValue={data.title}
+      />
+
+      <fieldset className="flex flex-row font-body mb-32">
+        <legend className="text-label text-btnBorder mb-16">
+          IS THE VIDEO YOURS OR SOMEONE ELSES?
```
Comment (dem4ron) [inline app/javascript/components/modals/upload-video/elements/UploadVideoForm.tsx:79]:
```suggestion
          IS THE VIDEO YOURS OR SOMEONE ELSE'S?
```

**Pull request merged**